prov/verbs/src/verbs_mr.c

```diff
@@ -272,9 +238,72 @@ vrb_mr_cache_reg(struct fid *fid, const void *buf, size_t len,
 	return FI_SUCCESS;
 }
 
-struct fi_ops_mr vrb_mr_cache_ops = {
+static int
+vrb_mr_reg_iface(struct fid *fid, const void *buf, size_t len, uint64_t access,
+		 uint64_t offset, uint64_t requested_key, uint64_t flags,
+		 struct fid_mr **mr, void *context, enum fi_hmem_iface iface)
+{
+	struct vrb_domain *domain;
+
+	domain = container_of(fid, struct vrb_domain,
+			      util_domain.domain_fid.fid);
+
+	if (domain->cache.monitors[iface])
+		return vrb_mr_cache_reg(domain, buf, len, access, offset,
+					requested_key, flags, mr, context,
+					iface);
+	else
+		return vrb_mr_nocache_reg(domain, buf, len, access, offset,
+					  requested_key, flags, mr, context,
+					  iface);
+}
+
+static int
+vrb_mr_regv_iface(struct fid *fid, const struct iovec *iov, size_t count,
+		  uint64_t access, uint64_t offset, uint64_t requested_key,
+		  uint64_t flags, struct fid_mr **mr, void *context,
+		  enum fi_hmem_iface iface)
+{
+	const void *addr = count ? iov->iov_base: NULL;
+	size_t len = count ? iov->iov_len : 0;
+
+	if (OFI_UNLIKELY(count > 1))
+		return -FI_EINVAL;
+
+	return vrb_mr_reg_iface(fid, addr, len, access, offset, requested_key,
+				flags, mr, context, iface);
+}
+
+static int
+vrb_mr_reg(struct fid *fid, const void *buf, size_t len, uint64_t access,
+	   uint64_t offset, uint64_t requested_key, uint64_t flags,
+	   struct fid_mr **mr, void *context)
+{
+	return vrb_mr_reg_iface(fid, buf, len, access, offset, requested_key,
+				flags, mr, context, FI_HMEM_SYSTEM);
+}
+
+static int
+vrb_mr_regv(struct fid *fid, const struct iovec *iov, size_t count,
+	    uint64_t access, uint64_t offset, uint64_t requested_key,
+	    uint64_t flags, struct fid_mr **mr, void *context)
+{
+	return vrb_mr_regv_iface(fid, iov, count, access, offset, requested_key,
+				 flags, mr, context, FI_HMEM_SYSTEM);
+}
+
+static int vrb_mr_regattr(struct fid *fid, const struct fi_mr_attr *attr,
+			  uint64_t flags, struct fid_mr **mr)
+{
+	return vrb_mr_regv_iface(fid, attr->mr_iov, attr->iov_count,
+				 attr->access, attr->offset,
+				 attr->requested_key, flags, mr, attr->context,
+				 attr->iface);
+}
```

Comment:
I know this PR has been deferred, but see PR #6152.  We need to ensure that verbs maintains backwards compatibility with existing apps.  You can test this by building fabtests against a 1.9 release and running them against this one.

Reply:
I'll verify and get back to you on this.

Reply:
I did not hit an issue, but the ABI check from https://github.com/ofiwg/libfabric/pull/6152 seems to be needed here as well. I'll update this PR with the ABI check from https://github.com/ofiwg/libfabric/pull/6152.